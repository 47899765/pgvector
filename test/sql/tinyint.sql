SELECT '127'::tinyint;
SELECT '128'::tinyint;
SELECT '-128'::tinyint;
SELECT '-129'::tinyint;

SELECT ''::tinyint;
SELECT ' 1'::tinyint;
SELECT '1 '::tinyint;
SELECT '1a'::tinyint;

SELECT '{1,2,3}'::tinyint[];

SELECT '128'::numeric::tinyint;
SELECT 'NaN'::numeric::tinyint;

SELECT l2_distance('{0,0}'::tinyint[], '{3,4}'::tinyint[]);
SELECT l2_distance('{0,0}'::tinyint[], '{0,1}'::tinyint[]);
SELECT l2_distance('{1,2}'::tinyint[], '{3}'::tinyint[]);
SELECT l2_distance('{3e38}'::tinyint[], '{-3e38}'::tinyint[]);
SELECT '{0,0}'::tinyint[] <-> '{3,4}'::tinyint[];

SELECT inner_product('{1,2}'::tinyint[], '{3,4}'::tinyint[]);
SELECT inner_product('{1,2}'::tinyint[], '{3}'::tinyint[]);
SELECT inner_product('{127}'::tinyint[], '{127}'::tinyint[]);
SELECT '{1,2}'::tinyint[] <#> '{3,4}'::tinyint[];

SELECT cosine_distance('{1,2}'::tinyint[], '{2,4}'::tinyint[]);
SELECT cosine_distance('{1,2}'::tinyint[], '{0,0}'::tinyint[]);
SELECT cosine_distance('{1,1}'::tinyint[], '{1,1}'::tinyint[]);
SELECT cosine_distance('{1,0}'::tinyint[], '{0,2}'::tinyint[]);
SELECT cosine_distance('{1,1}'::tinyint[], '{-1,-1}'::tinyint[]);
SELECT cosine_distance('{1,2}'::tinyint[], '{3}'::tinyint[]);
SELECT cosine_distance('{3e38}'::tinyint[], '{3e38}'::tinyint[]);
SELECT '{1,2}'::tinyint[] <=> '{2,4}'::tinyint[];
