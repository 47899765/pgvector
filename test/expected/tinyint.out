SELECT '127'::tinyint;
 tinyint 
---------
 127
(1 row)

SELECT '128'::tinyint;
ERROR:  value "128" is out of range for type tinyint
LINE 1: SELECT '128'::tinyint;
               ^
SELECT '-128'::tinyint;
 tinyint 
---------
 -128
(1 row)

SELECT '-129'::tinyint;
ERROR:  value "-129" is out of range for type tinyint
LINE 1: SELECT '-129'::tinyint;
               ^
SELECT ''::tinyint;
ERROR:  invalid input syntax for type tinyint: ""
LINE 1: SELECT ''::tinyint;
               ^
SELECT ' 1'::tinyint;
 tinyint 
---------
 1
(1 row)

SELECT '1 '::tinyint;
 tinyint 
---------
 1
(1 row)

SELECT '1a'::tinyint;
ERROR:  invalid input syntax for type tinyint: "1a"
LINE 1: SELECT '1a'::tinyint;
               ^
SELECT '{1,2,3}'::tinyint[];
 tinyint 
---------
 {1,2,3}
(1 row)

SELECT '128'::numeric::tinyint;
ERROR:  value "128" is out of range for type tinyint
SELECT 'NaN'::numeric::tinyint;
ERROR:  cannot convert NaN to integer
SELECT l2_distance('{0,0}'::tinyint[], '{3,4}'::tinyint[]);
 l2_distance 
-------------
           5
(1 row)

SELECT l2_distance('{0,0}'::tinyint[], '{0,1}'::tinyint[]);
 l2_distance 
-------------
           1
(1 row)

SELECT l2_distance('{1,2}'::tinyint[], '{3}'::tinyint[]);
 l2_distance 
-------------
            
(1 row)

SELECT l2_distance('{3e38}'::tinyint[], '{-3e38}'::tinyint[]);
ERROR:  invalid input syntax for type tinyint: "3e38"
LINE 1: SELECT l2_distance('{3e38}'::tinyint[], '{-3e38}'::tinyint[]...
                           ^
SELECT '{0,0}'::tinyint[] <-> '{3,4}'::tinyint[];
 ?column? 
----------
        5
(1 row)

SELECT inner_product('{1,2}'::tinyint[], '{3,4}'::tinyint[]);
 inner_product 
---------------
            11
(1 row)

SELECT inner_product('{1,2}'::tinyint[], '{3}'::tinyint[]);
 inner_product 
---------------
              
(1 row)

SELECT inner_product('{127}'::tinyint[], '{127}'::tinyint[]);
 inner_product 
---------------
         16129
(1 row)

SELECT '{1,2}'::tinyint[] <#> '{3,4}'::tinyint[];
 ?column? 
----------
      -11
(1 row)

SELECT cosine_distance('{1,2}'::tinyint[], '{2,4}'::tinyint[]);
 cosine_distance 
-----------------
               0
(1 row)

SELECT cosine_distance('{1,2}'::tinyint[], '{0,0}'::tinyint[]);
 cosine_distance 
-----------------
             NaN
(1 row)

SELECT cosine_distance('{1,1}'::tinyint[], '{1,1}'::tinyint[]);
 cosine_distance 
-----------------
               0
(1 row)

SELECT cosine_distance('{1,0}'::tinyint[], '{0,2}'::tinyint[]);
 cosine_distance 
-----------------
               1
(1 row)

SELECT cosine_distance('{1,1}'::tinyint[], '{-1,-1}'::tinyint[]);
 cosine_distance 
-----------------
               2
(1 row)

SELECT cosine_distance('{1,2}'::tinyint[], '{3}'::tinyint[]);
 cosine_distance 
-----------------
                
(1 row)

SELECT cosine_distance('{3e38}'::tinyint[], '{3e38}'::tinyint[]);
ERROR:  invalid input syntax for type tinyint: "3e38"
LINE 1: SELECT cosine_distance('{3e38}'::tinyint[], '{3e38}'::tinyin...
                               ^
SELECT '{1,2}'::tinyint[] <=> '{2,4}'::tinyint[];
 ?column? 
----------
        0
(1 row)

